src/enclave_proc/commands.rs

```diff
@@ -46,13 +47,28 @@ pub fn run_enclaves(
     .map_err(|e| {
         e.add_subaction("Failed to construct EnclaveManager with given arguments".to_string())
     })?;
+
+    // Launch asynchronous computing of PCRs
+    let mut async_manager = enclave_manager.clone();
+    let act = async_manager.get_pcrs(connection);
+
     enclave_manager
         .run_enclave(connection)
         .map_err(|e| e.add_subaction("Failed to run enclave".to_string()))?;
     enclave_manager
         .update_state(EnclaveState::Running)
         .map_err(|e| e.add_subaction("Failed to update enclave state".to_string()))?;
 
+    let runtime = Runtime::new().map_err(|e| {
+        new_nitro_cli_failure!(
+            &format!("Failed to initialize runtime: {:?}", e),
+            NitroCliErrorEnum::EnclaveBootFailure
+        )
+    })?;
+    runtime
+        .block_on(act)
```

Comment:
If you block_on here, you are definitely keeping the thread busy here don't you ?

Reply:
Yes, but it's blocking after what I assumed to be the more time consuming part, calling the function 'run_enclave' which allocates resources and starts the enclave.

Reply:
Modified, the thread is no longer blocking in run_enclaves function.